azure-pipelines.yml

```diff
@@ -0,0 +1,69 @@
+# nimi-bot pipeline for running all system tests as part of Continuous Integration.
+
+parameters:
+- name: module_names
+  type: object
+  default: # Order of the modules is based on test execution time, longest firt.
+  - nidigital
+  - nitclk  
+  - nifgen
+  - nidcpower
+  - nidmm
+  - niscope
+  - nimodinst
+  - nise
+  - niswitch
+
+# A pipeline with no CI trigger
+trigger: none
+
+# specific path
+pr:
+  branches:
+    include:
+    - master
+    - releases/*
+  paths:
+    exclude:
+    - CHANGELOG.md
+    - CONTRIBUTING.md
+    - .gitattributes
+    - LICENSE
+    - VERSION
+  drafts: false
+
+stages:
+
+  - stage: ensure_codegen_up_to_date
+    pool:
+      name: nimi-bot
+      demands:
+        - nimi-bot -equals True
+        - agent.os -equals Linux
+    jobs:
+    - job: ensure_codegen_up_to_date
+      workspace:
+        clean: all
+      steps:
+        - checkout: self  
+          clean: true
+          fetchDepth: $(clone_depth)
+        - task: PythonScript@0
+          displayName: Ensure changes to code generation are committed to repository
+          inputs:
+            scriptSource: filePath
+            scriptPath: tools/ensure_codegen_up_to_date.py
+            
+  - stage: system_tests_win64
+    dependsOn: ensure_codegen_up_to_date
+    pool:
+      name: nimi-bot
+      demands:
+        - nimi-bot -equals True
+        - agent.os -equals Windows_NT
+    jobs:
+    - ${{each module_name in parameters.module_names}}:
+      - template: .template_system_tests.yml
+        parameters:
+          module_name: ${{module_name}}
+          os_name: win64
```

Comment:
the fact that agent.os is "Linux" or "Windows_NT" makes me feel even more sure that `os_name` is not a good name for this parameter :)

Reply:
As suggested changed to platform